threedi_schema/application/schema.py

```diff
@@ -250,46 +236,70 @@ def convert_to_geopackage(self):
             with work_db.get_session() as session:
                 session.execute(text("DROP TABLE IF EXISTS spatialite_history;"))
                 session.execute(text("DROP TABLE IF EXISTS views_geometry_columns;"))
-            cmd = [
-                "ogr2ogr",
-                "-skipfailures",
-                "-f",
-                "gpkg",
-                str(Path(self.db.path).with_suffix(".gpkg")),
-                str(work_db.path),
-                "-oo",
-                "LIST_ALL_TABLES=YES",
-            ]
-            try:
-                p = subprocess.Popen(
-                    cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=-1
+
+                all_tablenames = [model.__tablename__ for model in self.declared_models]
+                geometry_tablenames = (
+                    session.execute(text("SELECT f_table_name FROM geometry_columns;"))
+                    .scalars()
+                    .all()
+                )
+                non_geometry_tablenames = [
+                    name for name in all_tablenames if name not in geometry_tablenames
+                ]
```

Comment:
I really like the simplicity of using sets in such cases

```suggestion
                non_geometry_tablenames = list(set(all_tablenames)-set(geometry_tablenames))
```